novawallet/Modules/Staking/StakingSetupAmount/Model/StakingTypeBalanceFactory.swift

```diff
@@ -0,0 +1,50 @@
+import Foundation
+import BigInt
+
+protocol StakingTypeBalanceFactoryProtocol: AnyObject {
+    func getAvailableBalance(
+        from assetBalance: AssetBalance?,
+        stakingMethod: StakingSelectionMethod
+    ) -> BigUInt?
+}
+
+final class StakingTypeBalanceFactory: StakingTypeBalanceFactoryProtocol {
+    let stakingType: StakingType?
+
+    init(stakingType: StakingType?) {
+        self.stakingType = stakingType
+    }
+
+    var stakingTypeAllowsLocks: Bool {
+        switch stakingType {
+        case .relaychain, .auraRelaychain, .azero, .none, .parachain, .turing:
+            return true
+        case .nominationPools, .unsupported:
+            return true
```

Comment:
is it correct?

Reply:
No, but I have fixed this in a separate PR